collect_app/src/main/java/org/odk/collect/android/formentry/saving/FormSaveViewModel.java

```diff
@@ -171,7 +173,7 @@ public String getReason() {
     }
 
     private void saveToDisk(SaveRequest saveRequest) {
-        saveTask = new SaveTask(saveRequest, formSaver, formController, mediaManager, new SaveTask.Listener() {
+        saveTask = new SaveTask(saveRequest, formSaver, formController, this, new SaveTask.Listener() {
```

Comment:
Hmmm. Little worried about the viewmodel being passed into an async task here... It's another awkward change but I think really what should happen here is that a list of files to save should get passed into the save task. Does that make sense? In that world we wouldn't need `saveChanges` (as it would be handled in `DiskFormSaver`.